app/controllers/api/base_controller.rb

```diff
@@ -54,5 +53,50 @@ def set_access_control_headers
     def collection_config
       @collection_config ||= CollectionConfig.new
     end
+
+    def log_request_initiated
+      @requested_at = Time.now.utc
+      $api_log.info { " " }
+      $api_log.info do
+        format_data_for_logging("API Request",
+                                :requested_at => @requested_at.to_s,
+                                :method       => request.request_method,
+                                :url          => request.original_url)
+      end
+    end
+
+    def log_api_request
+      @parameter_filter ||= ActionDispatch::Http::ParameterFilter.new(Rails.application.config.filter_parameters)
+      $api_log.info { format_data_for_logging("Request", @req.to_hash) }
+
+      $api_log.info do
+        unfiltered_params = request.query_parameters
+                                   .merge(params.permit(:action, :controller, :format).to_h)
+                                   .merge("body" => @req.json_body)
+        format_data_for_logging("Parameters", @parameter_filter.filter(unfiltered_params))
+      end
+      log_request_body
+    end
+
+    def log_request_body
+      if @req.json_body.present?
+        $api_log.debug { format_data_for_logging("Body", JSON.pretty_generate(@req.json_body)) }
+      end
+    end
+
+    def log_api_response
+      @completed_at = Time.now.utc
+      $api_log.info do
+        format_data_for_logging("Response",
+                                :completed_at => @completed_at.to_s,
+                                :size         => '%.3f KBytes' % (response.body.size / 1000.0),
+                                :time_taken   => '%.3f Seconds' % (@completed_at - @requested_at),
+                                :status       => response.status)
+      end
+    end
+
+    def format_data_for_logging(header, data)
+      "#{('%s:' % header).ljust(15)} #{data}"
+    end
```

Comment:
There is quite a good example [here](https://github.com/ManageIQ/manageiq/blob/master/spec/requests/api/logging_spec.rb#L60-L62)